editor/src/core/vscode/vscode-bridge.ts

```diff
@@ -95,12 +95,18 @@ export function initVSCodeBridge(
   openFilePath: string | null,
 ) {
   let loadingScreenHidden = false
+  let seenMessageListenersReadyMessage = false
 
   // Remove any existing message handlers to prevent us accidentally duplicating them
   window.removeEventListener('message', registeredHandlers)
   registeredHandlers = (messageEvent: MessageEvent) => {
     const { data } = messageEvent
-    if (isMessageListenersReady(data) && messageEvent.source != null) {
+    if (
+      isMessageListenersReady(data) &&
+      messageEvent.source != null &&
+      !seenMessageListenersReadyMessage
```

Comment:
should this throw an error if multiple ready messages arrive?

Reply:
I don't _think_ so, possibly more than one could end up in the pipeline, it's not really an error state just something we need to protect against.